R/prepare_data.R

```diff
@@ -53,10 +53,15 @@ prepare_data <- function(polygon_shapefile,
   
   polygon_data <- getPolygonData(polygon_shapefile, id_var = id_var, response_var = response_var)
   
+
+  # Save raster layer names so we can reassign it to make sure names don't change.
+  cov_names <- names(covariate_rasters)
+
   # If no aggregation raster is given, use a 'unity' raster
   if(is.null(aggregation_raster)) {
     aggregation_raster <- covariate_rasters[[1]]
-    raster::setValues(aggregation_raster, rep(1, raster::ncell(aggregation_raster)))
+    aggregation_raster <- raster::setValues(aggregation_raster, rep(1, raster::ncell(aggregation_raster)))
+    names(aggregation_raster) <- 'aggregation_raster'
```

Comment:
Should setting the name of the aggregation raster be outside the if statement. So it has the same name whether it was user defined or not?

Reply:
Yes probably.